src/nvim/drawline.c

```diff
@@ -1551,6 +1551,13 @@ int win_line(win_T *wp, linenr_T lnum, int startrow, int endrow, bool number_onl
       }
       // don't clear anything after wlv.col
       win_put_linebuf(wp, wlv.row, 0, wlv.col, wlv.col, bg_attr, false);
+      // update the 'statuscolumn' for the entire line size
+      if (number_only && statuscol.draw
+          && ++wlv.row - wlv.startrow < wp->w_lines[wlv.lnum - wp->w_topline].wl_size) {
```

Comment:
Hmm this `w_lines` index breaks down with folds. Can we pass `idx` from `win_update()` to `win_line()`?

Reply:
@zeertzjq @bfredl any thoughts on the (already pushed) `number_only` refactor here?